setup.py

```diff
@@ -31,6 +31,14 @@ def run_tests(self):
         self.run_command('flake8')
         sys.exit(errno)
 
+test_req = [
```

Comment:
I would suggest extracting both require and test require to requirements.txt and requirements-test.txt but probably in a separate PR. 
Having them in txt would help when doing releases

Reply:
thats kinda why i did these changes.  You can now do `pip install .[test]` to install test requirements